Add foolbox

beck-thompson opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->

Add the foolbox! The clowns favorite box filled with various items to help them on their pranking journey.

From: https://github.com/moff-station/moff-station-14/pull/160

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->

Cute item that helps fill their formally empty locker

## Technical details
<!-- Summary of code changes for easier review. -->

Added a filled version of the clown locker I hope mappers map it

## Media
<!-- Attach media if the PR makes in-game changes (clothing, items, features, etc).
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->

https://github.com/user-attachments/assets/08073a18-5447-441b-932d-d06145647182

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->

:cl: captainkawaii666
- add: A foolbox with various pranking items for clowns
- tweak: Crayon boxes are now slightly smaller

slarticodefast reviewed (changes requested):
Due to vital game balance reasons I request that you set the storage open and insert sounds correctly

<img width="465" height="277" alt="grafik" src="https://github.com/user-attachments/assets/1dde05e7-133f-4473-8948-387ab00fdf2d" />

beck-thompson commented:
> Due to vital game balance reasons I request that you set the storage open and insert sounds correctly

Added them! I added it for EVERY interaction so it might be too much let me know!

SnappingOpossum commented on Resources/Prototypes/Catalog/Fills/Items/toolboxes.yml:
```diff
@@ -157,3 +157,70 @@
+        children:
+        - id: RevolverCapGun
+        - id: BoxCartridgeCap
+          prob: 0.70
```
Above section could additionally be accomplished by doubling the weight of this entry, removing the above entry, and halving the prob of BoxCartridgeCap, or by deceasing this ones weight to 0.7 and removing the prob but increasing the above entry's weight to 1.3.

iaada commented on Resources/Prototypes/Entities/Objects/Tools/toolbox.yml:
```diff
@@ -147,3 +147,36 @@
+      params:
+        variation: 0.03
+        volume: 3
+
```
Extra line

iaada commented on Resources/Prototypes/Catalog/Fills/Items/toolboxes.yml:
```diff
@@ -157,3 +157,61 @@
+      - id: ClothingHandsGlovesConducting
+        weight: 0.0025
+      - id: EnergySword
+        weight: 0.0005
```
I think weights this small are helped by having a comment unwrapping the actual chance.

iaada commented on Resources/Prototypes/Catalog/Fills/Items/toolboxes.yml:
```diff
@@ -157,3 +157,75 @@
+    - !type:GroupSelector # Fun clothing
+      prob: 0.33
+      children:
+      - id: ClothingHandsGlovesColorYellowBudget
```
You don't need to change the chances, but the yellow gloves being together would read better.

Hmm though, they are grouped differently. Maybe a RandomYellowGloveTable should be used? There's a lot of places that could benefit from it.

iaada commented on Resources/Prototypes/Catalog/Fills/Items/toolboxes.yml:
```diff
@@ -157,3 +157,75 @@
+      - id: FoodBanana
+      - id: FoodPieBananaCream
+        weight: 0.75
+    - !type:GroupSelector # Fun clothing
```
I spawned a few thousand foolboxes and it seems like it'll be fine to have 2x2s in the clothing. There's a lot of outer clothing that would fit, some of which is literally not used anywhere. Like ClothingOuterCoatSpaceAsshole, though it might be a bad idea to give that to clowns. :o)

Ideas that I found glancing: ClothingOuterCoatExpensive, ClothingOuterGhostSheet, ClothingShoesBling. Another idea is to have an item for each department to encourage ~~pretending to help~~ getting a new job. ClothingOuterVestHazard, ClothingOuterCoatLab, ClothingMaskClownSecurity, ClothingHeadHatCargosoft.

beck-thompson replied:
Let me know what you think :saluting_face:

Hitlinemoss commented on Resources/Prototypes/Catalog/Fills/Lockers/service.yml:
```diff
@@ -182,3 +182,19 @@
     - id: ClothingOuterSuitJanitorBomb
+
+- type: entity
+  id: LockerClownFilled
```
Something I want to make note of here is that this is a different prototype than the existing clown locker, so maps will need to be manually updated to actually replace those lockers. I'll also note that some maps don't have even the current empty locker(s) mapped anywhere (namely Plasma, I think maybe at least one other but I forget off the top of my head), so that's something that'll also need to be addressed. Swapping the lockers out should be their own individual per-map PRs, though (ideally alongside other small updates/fixes to the map in the same PR), since that's how mapping changes are generally done for these kinds of things.

I do have my own PR open (#40122) for putting some things in the clown/mime lockers based on what items are near-universally found mapped in their room(s), but once these lockers _are_ mapped, changing their contents is just a matter of changing what's listed in their fill tables, so the contents don't need to be 100% decided on before they actually get mapped or anything. I mostly just mention all this as a sort of reminder that we need to map this new locker eventually.

iaada reviewed (approved):
It all looks good to me! Let the clown revolution begin :o)
